sql/atxs/atxs.go

```diff
@@ -926,3 +926,40 @@ func SetUnits(db sql.Executor, atxID types.ATXID, id types.NodeID, units uint32)
 	)
 	return err
 }
+
+// AtxWithPrevious returns the ATX ID that has the given ATX ID as its previous ATX.
+func AtxWithPrevious(db sql.Executor, prev types.ATXID, id types.NodeID) (types.ATXID, error) {
+	var (
+		atxid types.ATXID
+		rows  int
+		err   error
+	)
+	decode := func(s *sql.Statement) bool {
+		s.ColumnBytes(0, atxid[:])
+		return false
+	}
+	if prev == types.EmptyATXID {
+		rows, err = db.Exec("SELECT id FROM atxs WHERE pubkey = ?1 AND prev_id IS NULL;",
```

Comment:
I think it makes sense to still add sorting by received to the query to ensure it will always return the same ATX if there are multiple ATXs with the same `prev`:

```suggestion
		rows, err = db.Exec(`
			SELECT id FROM atxs
			WHERE pubkey = ?1 AND prev_id IS NULL
			ORDER BY received ASC;`
		)
```